tasks/orca-recovery-adapter/src/index.ts

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+import pRetry from 'p-retry';
+import { v4 as uuidv4 } from 'uuid';
+import { Context } from 'aws-lambda';
+
+import { sfn } from '@cumulus/aws-client/services';
+import { describeExecution } from '@cumulus/aws-client/StepFunctions';
+import { runCumulusTask, CumulusMessageWithAssignedPayload } from '@cumulus/cumulus-message-adapter-js';
+import Logger from '@cumulus/logger';
+import { buildExecutionArn } from '@cumulus/message/Executions';
+import { CumulusMessage, CumulusRemoteMessage } from '@cumulus/types/message';
+
+import { HandlerEvent, HandlerOutput } from './types';
+
+const log = new Logger({ sender: '@cumulus/orca-recovery-adapter' });
+
+/**
+ * Get state machine execution results
+ *
+ * @param {object} params
+ * @param {string} params.executionArn - execution arn
+ * @param {number} params.retries - number of retries
+ * @param {number} params.retryIntervalInSecond - retry internal in second
+ * @param {number} params.maxRetryTimeInSecond - max retry time in second
+ * @returns Promise<AWS.StepFunctions.DescribeExecutionOutput> - Returns promise
+ *   that resolves to the output of step function execution
+ */
+export const getStateMachineExecutionResults = async (
+  params: {
+    executionArn: string,
+    retries?: number,
+    retryIntervalInSecond?: number,
+    maxRetryTimeInSecond?: number,
+  }
+) : Promise<AWS.StepFunctions.DescribeExecutionOutput> => {
+  const {
+    executionArn,
+    retries = 50,
+    retryIntervalInSecond = 5,
+    maxRetryTimeInSecond = 1800,
+  } = params;
+  const result = await pRetry(
```

Comment:
```suggestion
  return await pRetry(
```
nit: return directly